app/views/exam_templates/_split_pdf_log.html.erb

```diff
@@ -0,0 +1,28 @@
+<div>
+  <% current_incomplete_num_groups = split_pdf_log.num_groups_in_incomplete %>
+  <% if current_incomplete_num_groups == previous_incomplete_num_groups && split_pdf_log.num_pages_qr_scan_error == 0 %>
+    <p class="log-green-message">
+      <%= "#{t('exam_templates.split_pdf_log.log')} ##{i}" %> - <%= split_pdf_log.uploaded_when.strftime("%d %b %y at %I:%M:%S %p") %>
+    </p>
+    <p><%= t('exam_templates.split_pdf_log.success_log', filename: split_pdf_log.filename, original_num_pages: split_pdf_log.original_num_pages) %>.</p>
+  <% else %>
+    <p class="log-red-message">
+      <%= "#{t('exam_templates.split_pdf_log.log')} ##{i}" %> - <%= split_pdf_log.uploaded_when.strftime("%d %b %y at %I:%M:%S %p") %>
+    </p>
+    <% if split_pdf_log.num_pages_qr_scan_error > 0 %>
+        <p><%= t('exam_templates.split_pdf_log.failure_general_log', filename: split_pdf_log.filename, original_num_pages: split_pdf_log.original_num_pages, num_pages_qr_scan_error: split_pdf_log.num_pages_qr_scan_error) %>.</p>
+    <% else %>
```

Comment:
You should have two independent `if` blocks here. Both error messages can appear.